infrastructure.md

```diff
@@ -8,4 +8,100 @@ The best description of the full infrastructure is currently given as an [Ansibl
+
+Where are they located? What python module do we use to access them?
+
+### Local file storages
+
+### Triple store
+
+## Running experiments
+
+Experiments are run by calling a `cgi` script, that calls the `fc-runner` (see below).
+
+
+### FC-Runner
+
+The [fc-runner](https://github.com/ModellingWebLab/fc-runner) repository contains a CGI script that can be called by the Django front-end.
+It handles three tasks:
+
+- Determining a protocol interface (i.e. the variable annotations it requires)
+- Scheduling experiments
+- Cancelling scheduled experiments
+
+### Scheduling experiments
+
+Experiments are schedule by placing them into a [Celery](http://docs.celeryproject.org/en/latest/index.html) task queue ([wiki](https://en.wikipedia.org/wiki/Celery_(software)).
+One or multiple workers then take tasks from the queue and execute them.
+This way, the actual running of experiments can be offloaded to a cloud of worker machines.
+
+The workers talk to the queue via a message passing system.
+Instead of talking to each other directly, messaging is handled via a *broker*, for which Web Lab uses [RabbitMQ](https://www.rabbitmq.com/documentation.html) ([wiki](https://en.wikipedia.org/wiki/RabbitMQ)).
+See also [Celery docs: Using RabbitMQ](http://docs.celeryproject.org/en/latest/getting-started/brokers/rabbitmq.html).
+
+Further information is given [here](./infrastructure-cardiac.md).
+
+## Deployment with Vagrant and Ansible
+
+[Vagrant](https://www.vagrantup.com/docs/index.html) ([wiki](https://en.wikipedia.org/wiki/Vagrant_%28software%29)) is a tool for "building and maintaining virtual software development environments".
+For the WebLab, we use Vagrant to create and manage [VirtualBox](https://en.wikipedia.org/wiki/VirtualBox) machines, which are automatically set up for development or production using Ansible.
+Users can connect to running vagrant machines using ``$ vagrant ssh``.
+
+[Ansible](https://docs.ansible.com/) ([wiki](https://en.wikipedia.org/wiki/Ansible_(software))) is a tool to set up development environments, or deploy applications to some production environment.
+
+The WebLab [deployment](https://github.com/ModellingWebLab/deployment) repo contains several [Ansible playbooks](https://docs.ansible.com/ansible/latest/user_guide/playbooks_intro.html), which each set up some part of the WebLab infrastructure.
+Overlapping parts of playbooks are shared via [roles](https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html)).
+
+See also: [Using Vagrant and Ansible](https://docs.ansible.com/ansible/latest/scenario_guides/guide_vagrant.html)
+
+Five playbooks are defined:
+
+### Site
+
+Stored in `site.yml`, this is the main playbook, and does nothing other than simply importing the remaining four playbooks.
+
+### Web servers
+
+Stored in `webservers.yml`, this sets up a Django server.
+**JONATHAN: Can there be more than one of these as the name suggests? How would they interact? Esp. given local files?**
```

Comment:
At present just one in a given Web Lab. But we might potentially want to consider load balancing if a public version got big!